tools/xtce_generation/telemetry_generator.py

```diff
@@ -0,0 +1,294 @@
+
+        Parameters:
+        - parameter_set: The ParameterSet element where parameters will be added.
+        - ccsds_parameters: A list of dictionaries containing CCSDS parameter data.
+
+        Returns:
+        - parameter_set: The updated ParameterSet element.
+        """
+        for parameter_data in ccsds_parameters:
+            parameter = Et.SubElement(parameter_set, "xtce:Parameter")
+            parameter.attrib["name"] = parameter_data["name"]
+            parameter.attrib["parameterTypeRef"] = parameter_data["parameterTypeRef"]
+
+            description = Et.SubElement(parameter, "xtce:LongDescription")
+            description.text = parameter_data["description"]
+
+        return parameter_set
+
+    def create_container_set(self, telemetry_metadata, ccsds_parameters, apid):
+        """
+        Create XML elements for ContainerSet, CCSDSPacket SequenceContainer,
+        and CoDICESciencePacket SequenceContainer.
+
+        Parameters:
+        - telemetry_metadata: The TelemetryMetaData element where containers are.
+        - ccsds_parameters: A list of dictionaries containing CCSDS parameter data.
+        - APID: The APID value used for comparison in CoDICESciencePacket.
+
+        Returns:
+        - telemetry_metadata: The updated TelemetryMetaData element.
+        """
+        # Create ContainerSet element
+        container_set = Et.SubElement(telemetry_metadata, "xtce:ContainerSet")
+
+        # Create CCSDSPacket SequenceContainer
+        ccsds_packet_container = Et.SubElement(container_set, "xtce:SequenceContainer")
+        ccsds_packet_container.attrib["name"] = "CCSDSPacket"
+        ccsds_packet_entry_list = Et.SubElement(
+            ccsds_packet_container, "xtce:EntryList"
+        )
+
+        for parameter_data in ccsds_parameters:
+            parameter_ref_entry = Et.SubElement(
+                ccsds_packet_entry_list, "xtce:ParameterRefEntry"
+            )
+            parameter_ref_entry.attrib["parameterRef"] = parameter_data["name"]
+
+        # Create CoDICESciencePacket SequenceContainer
+        codice_science_container = Et.SubElement(
+            container_set, "xtce:SequenceContainer"
+        )
+        codice_science_container.attrib["name"] = "CoDICESciencePacket"
+
+        base_container = Et.SubElement(codice_science_container, "xtce:BaseContainer")
+        base_container.attrib["containerRef"] = "CCSDSPacket"
+
+        restriction_criteria = Et.SubElement(base_container, "xtce:RestrictionCriteria")
+        comparison = Et.SubElement(restriction_criteria, "xtce:Comparison")
+        comparison.attrib["parameterRef"] = "PKT_APID"
+        comparison.attrib["value"] = apid
+        comparison.attrib["useCalibratedValue"] = "false"
+
+        Et.SubElement(codice_science_container, "xtce:EntryList")
+
+        return telemetry_metadata
+
+    def add_codice_science_parameters(self, codice_science_container, pkt):
```

Comment:
Is this codice specific? It looks pretty generic to me. If you move this to "science_parameters" and "science_container", then this whole class becomes non-instrument specific. 

That being said, maybe this is exclusively codice, in which case you can leave it as is :)